### PR TITLE
Allow specifying read consistency

### LINKED ISSUE #4340: Datastore: please support specifying read consistency
According to the Cloud Datastore docs, the python client does not support setting the read consistency of a query:
https://cloud.google.com/datastore/docs/concepts/structuring_for_strong_consistency#datastore-eventual-consistent-query-python

Any chance this could be implemented?

### PR DESCRIPTION
**Pull request by chemelnucfin**
Closes #4340

### CODE REVIEW

**Comment by chemelnucfin** [comment]
Ok.

**Comment by tseaver** [comment]
@chemelnucfin Please put the "Closes #4340" in the description, not the title.  E.g., for a single-commit PR:

$ git commit datastore/ -m "Specify read consistency.

Closes $4340."

**Comment by tseaver** [comment]
Also, please add labels when you create the PR.

File: datastore/google/cloud/datastore/client.py [hunk 1]
```diff
@@ -15,23 +15,24 @@
...
 import os
 
-from google.cloud.proto.datastore.v1 import datastore_pb2 as _datastore_pb2
-
 from google.cloud._helpers import _LocalStack
-from google.cloud._helpers import (
-    _determine_default_project as _base_default_project)
+from google.cloud._helpers import (_determine_default_project as
+                                   _default_project)
+
```
Comment (dhermes) [inline datastore/google/cloud/datastore/client.py]: This empty line doesn't belong IIUC (it's pretty irrelevant, but I'm pointing it out for some reason)

File: datastore/google/cloud/datastore/client.py [hunk 2]
```diff
@@ -15,23 +15,24 @@
...
 
-from google.cloud.datastore._http import HTTPDatastoreAPI
 from google.cloud.datastore import helpers
+from google.cloud.datastore._http import HTTPDatastoreAPI
 from google.cloud.datastore.batch import Batch
 from google.cloud.datastore.entity import Entity
 from google.cloud.datastore.key import Key
 from google.cloud.datastore.query import Query
 from google.cloud.datastore.transaction import Transaction
+
```
Comment (dhermes) [inline datastore/google/cloud/datastore/client.py]: Ditto here on the empty line

File: datastore/google/cloud/datastore/client.py [hunk 3]
```diff
@@ -74,7 +75,7 @@ def _determine_default_project(project=None):
         project = _get_gcd_project()
 
     if project is None:
-        project = _base_default_project(project=project)
+        project = _default_project(project=project)
```
Comment (dhermes) [inline datastore/google/cloud/datastore/client.py]: I'm pretty unclear on why this (and the import) were changed at all. They have nothing to do with the PR AFAICT.

File: datastore/google/cloud/datastore/client.py [hunk 4]
```diff
@@ -302,15 +304,27 @@ def get(self, key, missing=None, deferred=None, transaction=None):
         :param transaction: (Optional) Transaction to use for read consistency.
                             If not passed, uses current transaction, if set.
 
+        :type eventual: bool
+        :param eventual: (Optional) Defaults to strongly consistent (False).
+                                    Setting True will use eventual consistency,
```
Comment (dhermes) [inline datastore/google/cloud/datastore/client.py]: "Setting" should align with "(Optional) ..."

File: datastore/google/cloud/datastore/client.py [hunk 5]
```diff
@@ -350,7 +371,8 @@ def get_multi(self, keys, missing=None, deferred=None, transaction=None):
         entity_pbs = _extended_lookup(
             datastore_api=self._datastore_api,
             project=self.project,
-            key_pbs=[k.to_protobuf() for k in keys],
+            key_pbs=[key.to_protobuf() for key in keys],
```
Comment (dhermes) [inline datastore/google/cloud/datastore/client.py]: Thanks for the `k` -> `key` change :grinning:

File: datastore/google/cloud/datastore/helpers.py [hunk 6]
```diff
@@ -19,16 +19,17 @@
...
-from google.type import latlng_pb2
 import six
 
 from google.cloud._helpers import _datetime_to_pb_timestamp
 from google.cloud._helpers import _pb_timestamp_to_datetime
-from google.cloud.proto.datastore.v1 import entity_pb2 as _entity_pb2
 from google.cloud.datastore.entity import Entity
 from google.cloud.datastore.key import Key
+from google.cloud.proto.datastore.v1 import entity_pb2 as _entity_pb2
+from google.cloud.proto.datastore.v1 import datastore_pb2 as _datastore_pb2
```
Comment (dhermes) [inline datastore/google/cloud/datastore/helpers.py]: You don't need to do this import alias-ing at this point. The reason the other ones are aliased are historical: there was a code change from `v1beta3` to `v1` and we didn't want to disturb the existing code.

File: datastore/google/cloud/datastore/client.py (hunk 3 above)
Comment (chemelnucfin) [reply datastore/google/cloud/datastore/client.py]: I think I was thinking I could get the import line to just under the 80 char limit.  But CircleCI was 79 chars.  I can change it back if you want.

Comment (dhermes) [reply datastore/google/cloud/datastore/client.py]: Please change it back. In general, a PR should **only** contain changes relevant to one thing being changed. (We break this from time to time of course, but should try very hard not to.)

File: datastore/google/cloud/datastore/helpers.py (hunk 6 above)
Comment (chemelnucfin) [reply datastore/google/cloud/datastore/helpers.py]: So just 
google.cloud.proto.datastore.v1 import entity_pb2
google.cloud.proto.datastore.v1 import datastore_pb2?

Comment (dhermes) [reply datastore/google/cloud/datastore/helpers.py]: Yes

**Comment by tseaver** [comment]
@chemelnucfin Rather than merging master into your branch (which is what confused the @goolebot) we prefer to rebase the branch against master, which keeps the PR focused on just the relevant changes.

**Comment by chemelnucfin** [comment]
Sorry about that.  I think I rebased on the wrong branch.  I think that's better now.

**Comment by chemelnucfin** [comment]
@dhermes @tseaver Is this ok?

File: datastore/google/cloud/datastore/client.py [hunk 7]
```diff
@@ -15,23 +15,21 @@
 
 import os
 
-from google.cloud.datastore_v1.proto import datastore_pb2 as _datastore_pb2
-
 from google.cloud._helpers import _LocalStack
-from google.cloud._helpers import (
-    _determine_default_project as _base_default_project)
+from google.cloud._helpers import (_determine_default_project as
+                                   _base_default_project)
```
Comment (dhermes) [inline datastore/google/cloud/datastore/client.py]: @chemelnucfin In the future, please don't make cosmetic changes like that that are unrelated to the PR.

File: datastore/google/cloud/datastore/client.py (hunk 4 above)
Comment (dhermes) [reply datastore/google/cloud/datastore/client.py]: @chemelnucfin In the future, please address all review comments (like this one).

File: datastore/google/cloud/datastore/helpers.py [hunk 8]
```diff
@@ -19,17 +19,18 @@
...
 from google.cloud._helpers import _datetime_to_pb_timestamp
 from google.cloud._helpers import _pb_timestamp_to_datetime
-from google.cloud.datastore_v1.proto import entity_pb2 as _entity_pb2
+from google.cloud.datastore_v1.proto import datastore_pb2
+from google.cloud.datastore_v1.proto import entity_pb2
 from google.cloud.datastore.entity import Entity
 from google.cloud.datastore.key import Key
 
+from google.protobuf import struct_pb2
+from google.type import latlng_pb2
```
Comment (dhermes) [inline datastore/google/cloud/datastore/helpers.py]: @chemelnucfin See PEP8 for import order (also, this is a cosmetic change unrelated to your PR).

Import sections should be (in order):

- Standard Library
- Third Party (i.e. from PyPI)
- First Party (i.e. from our own package)

The `google.protobuf` import is from [`protobuf`][1], the `google.type` import is from [`googleapis-common-protos`][2]. Hence they both belong in 3rd-party imports with `six`.

(I'm also sorry our `lint` tooling doesn't enforce this. Being a human linter is no fun for the reviewer or author.) 

[1]: https://pypi.org/project/protobuf/
[2]: https://pypi.org/project/googleapis-common-protos

**Pull request merged**